pkgs/applications/networking/instant-messengers/quaternion/default.nix

```diff
@@ -1,46 +1,47 @@
 { mkDerivation, stdenv, lib, fetchFromGitHub, cmake
-, qtbase, qtquickcontrols, qtkeychain, qtmultimedia, qttools
-, libqmatrixclient_0_5
-, libsecret
+, qtbase, qtquickcontrols, qtquickcontrols2, qtkeychain, qtmultimedia, qttools
+, libquotient, gnome3, libsecret
 }:
 
-let
-  generic = version: sha256: prefix: library: mkDerivation {
-    pname = "quaternion";
-    inherit version;
-
-    src = fetchFromGitHub {
-      owner = "QMatrixClient";
-      repo  = "Quaternion";
-      rev   = "${prefix}${version}";
-      inherit sha256;
-    };
-
-    buildInputs = [ qtbase qtmultimedia qtquickcontrols qtkeychain library libsecret ];
-
-    nativeBuildInputs = [ cmake qttools ];
+mkDerivation rec {
+  pname = "quaternion";
+  version = "0.0.9.4e";
 
-    postInstall = if stdenv.isDarwin then ''
-      mkdir -p $out/Applications
-      mv $out/bin/quaternion.app $out/Applications
-      rmdir $out/bin || :
-    '' else ''
-      substituteInPlace $out/share/applications/quaternion.desktop \
-        --replace 'Exec=quaternion' "Exec=$out/bin/quaternion"
-    '';
-
-    meta = with lib; {
-      description = "Cross-platform desktop IM client for the Matrix protocol";
-      homepage    = "https://matrix.org/docs/projects/client/quaternion.html";
-      license     = licenses.gpl3;
-      maintainers = with maintainers; [ peterhoeg ];
-      inherit (qtbase.meta) platforms;
-      inherit version;
-    };
+  src = fetchFromGitHub {
+    owner = "QMatrixClient";
+    repo = "Quaternion";
+    rev = "${version}";
+    sha256 = "sha256-2yEiILiitRPj2hCodUDM8UNVq8crb9nyX21ebuh5EEM=";
   };
 
-in rec {
-  quaternion     = generic "0.0.9.4c"     "12mkwiqqbi4774kwl7gha72jyf0jf547acy6rw8ry249zl4lja54" "" libqmatrixclient_0_5;
-
-  quaternion-git = quaternion;
+  buildInputs = [
+    qtbase
+    qtmultimedia
+    qtquickcontrols
+    qtquickcontrols2
+    qtkeychain
+    libquotient
+    gnome3.libsecret
```

Comment:
Seems strange that `libsecret` seems to have been replaced with `gnome3.libsecret`?

Reply:
we should change this, since `gnome3.libsecret` is now an alias of `libsecret` (moved to all-packages.nix)